h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -180,6 +180,112 @@ static int nVIFModelsInParallel(GLMParameters parms) {
     return 2; // same strategy as in CV of "big data" - run 2 models concurrently
   }
 
+  class GLMContributionsWithBackground extends ContributionsWithBackgroundFrameTask<GLMContributionsWithBackground> {
+    double[] _betas;
+    DataInfo _dinfo;
+    boolean _compact;
+    boolean _outputSpace;
+
+    GLMContributionsWithBackground(DataInfo dinfo, double[] betas, Key<Frame> frameKey, Key<Frame> backgroundFrameKey, boolean perReference, boolean compact, boolean outputSpace) {
+      super(frameKey, backgroundFrameKey, perReference);
+      _dinfo = dinfo;
+      _betas = betas;
+      _compact = compact;
+      _outputSpace = outputSpace;
+    }
+
+    @Override
+    public void map(Chunk[] cs, Chunk[] bgCs, NewChunk[] ncs) {
+      DataInfo.Row row = _dinfo.newDenseRow();
+      DataInfo.Row bgRow = _dinfo.newDenseRow();
+      int idx;
+      double[] result = MemoryManager.malloc8d(ncs.length - 1); // used only when output_format == compact
+      double transformationRatio = 1; // used for transforming to output space 
+      for (int j = 0; j < cs[0]._len; j++) {
+        _dinfo.extractDenseRow(cs, j, row);
+
+        for (int k = 0; k < bgCs[0]._len; k++) {
+          _dinfo.extractDenseRow(bgCs, k, bgRow);
+          Arrays.fill(result, 0);
+          double biasTerm = (_dinfo._intercept ? _betas[_betas.length - 1] : 0);
+          for (int i = 0; i < _betas.length - (_dinfo._intercept ? 1 : 0); i++) {
+            idx = _compact ? _dinfo.coefOriginalColumnIndices()[i] : i;
+            result[idx] += _betas[i] * (row.get(i) - bgRow.get(i));
+            biasTerm += _betas[i] * bgRow.get(i);
```

Comment:
I think I will postpone the improvements you suggested to some future PR as it would mainly speedup the use-case without StackedEnsemble and this PR is mainly for StackedEnsemble. 

In the use-case with SE, there is as many bias terms as are the rows in the background data set and I could speed it up a bit but It would require me to reorder the rows in all other algos as well so I don't need to sort the contribution frames in the SE (I check for the same order but don't sort).

I already had implemented some version of the marginal shap for GLM but I removed it so I don't add unnecessary complexity (It doesn't really bring much more info than the coefficients in the GLM case).

https://github.com/h2oai/h2o-3/pull/15734/commits/26045d4e7ed20e1734e2bb5b3ab0423029f484c4#diff-b5a14183b7510c835fd11228e8c99c5e00ad87592933bf2569b3934a7879218fR188-R213